reanalysis/utils.py

```diff
@@ -0,0 +1,382 @@
+"""
+a collection of classes and methods
+which may be shared across reanalysis components
+"""
+
+
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+from dataclasses import dataclass, is_dataclass
+from csv import DictReader
+from functools import cache
+import json
+import logging
+import re
+
+import cyvcf2
+from cloudpathlib import AnyPath
+from cyvcf2 import Variant
+
+
+# CompHetDict structure:
+# {
+#     sample: {
+#         gene: {
+#             variant: [variant, ...]
+#         }
+#     }
+# }
+# sample: string, e,g, CGP12345
+# gene: string, e.g. ENSG012345
+# variant: string, chr-pos-ref-alt
+CompHetDict = Dict[str, Dict[str, Dict[str, List[str]]]]
+
+InfoDict = Dict[str, Union[str, Dict[str, str]]]
+PanelAppDict = Dict[str, Dict[str, Union[str, bool]]]
+
+COMP_HET_VALUES: List[str] = ['sample', 'gene', 'id', 'chrom', 'pos', 'ref', 'alt']
+
+HOMREF: int = 0
+HETALT: int = 1
+UNKNOWN: int = 2
+HOMALT: int = 3
+
+# in cyVCF2, these ints represent HOMREF, and UNKNOWN
+BAD_GENOTYPES: Set[int] = {HOMREF, UNKNOWN}
+
+
+@dataclass
+class PedPerson:
```

Comment:
I think @vladsaveliev / @illusional have already built pedigree classes or are reusing the Somalier ones?